rundeckapp/webhooks/grails-app/services/webhooks/WebhookService.groovy

```diff
@@ -179,6 +179,13 @@ class WebhookService {
         }
     }
 
+    boolean importIsAllowed(Webhook hook, Map hookData) {
+        if(hook.authToken == hookData.authToken) return true
+        if(!hook.authToken && Webhook.countByAuthToken(hookData.authToken) == 0) return true
```

Comment:
an additional check needed is something like `AuthToken.countByToken(hookData.authToken)==0`